commitizen/tags.py

```diff
@@ -0,0 +1,214 @@
+from __future__ import annotations
+
+import re
+from collections.abc import Sequence
+from dataclasses import dataclass, field
+from functools import cached_property
+from string import Template
+from typing import TYPE_CHECKING, NamedTuple
+
+from typing_extensions import Self
+
+from commitizen import out
+from commitizen.defaults import DEFAULT_SETTINGS, Settings, get_tag_regexes
+from commitizen.git import GitTag
+from commitizen.version_schemes import (
+    DEFAULT_SCHEME,
+    InvalidVersion,
+    Version,
+    VersionScheme,
+    get_version_scheme,
+)
+
+if TYPE_CHECKING:
+    from commitizen.version_schemes import VersionScheme
+
+
+class VersionTag(NamedTuple):
+    """Represent a version and its matching tag form."""
+
+    version: str
+    tag: str
+
+
+@dataclass
+class TagRules:
```

Comment:
could you add a mini-example of how to use this class? 🙏🏻 

```
@dataclass
class TagRules:
    """
    Encapsulate tag-related rules.

    Can some information come from the settings? is it design to give flexbility?
   
    Example:

    ```py
    rules = TagRules.from_settings(...)
    # etc...
    ```

Thanks 🙏🏻

Reply:
I added a big one 👍🏼